Source/JavaScriptCore/runtime/JSMicrotask.cpp

```diff
@@ -79,14 +86,18 @@ void runJSMicrotask(JSGlobalObject* globalObject, MicrotaskIdentifier identifier
     if (UNLIKELY(handlerArguments.hasOverflowed()))
         return;
 
-    if (UNLIKELY(globalObject->hasDebugger()))
+    if (UNLIKELY(globalObject->hasDebugger())) {
         globalObject->debugger()->willRunMicrotask(globalObject, identifier);
+        scope.clearException();
```

Comment:
Shouldn't we also check for `TerminationException` here and return early if needed?  We need to be careful of accidentally clearing the `TerminationException` here.  Maybe make a lambda `clearExceptionIfNotTermination` and use that?  The lambda can return `true` for success, and `false` for failure to clear.  If it returns false (i.e. we've got a termination), return early.

Also, I'm not sure what to do about the inspector given that we called `willRunMicrotask` earlier.  Maybe ask the inspector folks about this.  Alternatively, we can also use a `DeferTerminationForAWhileScope` around the inspector calls and ensure that we don't have to deal with termination conditions.  Yes, I think this is the better thing to do.

Reply:
Done.